### PR TITLE
Unit tests

### PR DESCRIPTION
**Pull request by abdulwahidsharief**

### CODE REVIEW

File: tests/phpunit/tests/test-instrumentation.php [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    {
+        parent::setup();
+        $this->rzpInstrumentation = Mockery::mock('TrackPluginInstrumentation',
+            ['key_id', 'key_secret'])->makePartial()->shouldAllowMockingProtectedMethods();
+    }
+
+    public function testHooks()
+    {
+        $this->assertSame(10, has_action('activated_plugin', 'razorpayPluginActivated'));
+        $this->assertSame(10, has_action('deactivated_plugin', 'razorpayPluginDeactivated'));
```
Comment (sonyy-m) [inline tests/phpunit/tests/test-instrumentation.php:22]: This test case is failing for the hooks i have changed in last release
[https://github.com/razorpay/razorpay-woocommerce/blob/master/woo-razorpay.php#L37-L39
](https://github.com/razorpay/razorpay-woocommerce/blob/master/woo-razorpay.php#L37-L39
)

**Review by sonyy-m — CHANGES_REQUESTED** [review]

File: tests/phpunit/tests/test-formfields.php [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    {
+        $form_fields= $this->razorpayTests->{"form_fields"};
+
+        $this->assertSame('Key ID',$form_fields['key_id']['title']);
+
+        $this->assertSame('Key Secret',$form_fields['key_secret']['title']);
+
+        $this->assertSame('Payment Action',$form_fields['payment_action']['title']);
+
+        $this->assertSame('Enable/Disable',$form_fields['enabled']['title']);
```
Comment (sonyy-m) [inline tests/phpunit/tests/test-formfields.php:23]: give a space before 2nd parameter

File: tests/phpunit/tests/test-formfields.php [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+    public function setup():void
+    {
+        parent::setup();
+        $this->razorpayTests = new WC_Razorpay();
+    }
+
+    public function testformfields()
+    {
+        $form_fields= $this->razorpayTests->{"form_fields"};
```
Comment (sonyy-m) [inline tests/phpunit/tests/test-formfields.php:15]: give a space after variable name

File: tests/phpunit/util/class-util.php [hunk 4]
```diff
@@ -0,0 +1,151 @@
+<?php
+
+class Util {
+
+    static public function has_value ( $value, $array ) {
+        $has_value = false;
+
+        $callback = function  ( $v, $k ) use( $value, &$has_value ) {
```
Comment (sonyy-m) [inline tests/phpunit/util/class-util.php:8]: remove the spaces before 1st parameter and after 2nd parameter.
follow php coding guidelines and update the code

**Review by sonyy-m — CHANGES_REQUESTED** [review]

File: tests/phpunit/tests/test-instrumentation.php (hunk 1 above)
Comment (abdulwahidsharief) [reply tests/phpunit/tests/test-instrumentation.php:22]: @sonyy-m register_activation_hook and register_deactivation_hook will differ the hook name like activate_plugin_dir/plugin.php will change the hook name

**Review by sonyy-m — APPROVED** [review]
LGTM

**Review by ramth05 — APPROVED** [review]
LGTM

**Pull request merged**